packages/models/src/provider-api.ts

```diff
@@ -256,6 +256,72 @@ export function getProviderEndpoint(
 	}
 }
 
+/**
+ * Get the cheapest model for a given provider based on input + output pricing
+ */
+export function getCheapestModelForProvider(
+	provider: ProviderId,
+): string | null {
+	const availableModels = models
+		.filter((model) => model.providers.some((p) => p.providerId === provider))
+		.map((model) => ({
+			model: model.model,
+			provider: model.providers.find((p) => p.providerId === provider)!,
+		}))
+		.filter(
+			({ provider: providerInfo }) =>
+				providerInfo.inputPrice !== undefined &&
+				providerInfo.outputPrice !== undefined,
+		);
+
+	if (availableModels.length === 0) {
+		return null;
+	}
+
+	let cheapestModel = availableModels[0].provider.modelName;
+	let lowestPrice = Number.MAX_VALUE;
+
+	for (const { provider: providerInfo } of availableModels) {
+		const totalPrice =
+			(providerInfo.inputPrice! + providerInfo.outputPrice!) / 2;
+		if (totalPrice < lowestPrice) {
+			lowestPrice = totalPrice;
+			cheapestModel = providerInfo.modelName;
+		}
+	}
+
+	return cheapestModel;
+}
+
+/**
+ * Get the cheapest provider and model from a list of available model providers
+ */
+export function getCheapestFromAvailableProviders<
```

Comment:
do not redefine this function, instead import this from @openllm/models since we already have this???